src/engine/source/json/src/json.cpp

```diff
@@ -100,7 +100,19 @@ std::string Json::formatJsonPath(std::string_view dotPath, bool skipDot)
         // Replace . with /
         if (!skipDot)
         {
-            std::replace(std::begin(ptrPath), std::end(ptrPath), '.', '/');
+            for (auto pos = ptrPath.find('.'); pos != std::string::npos; pos = ptrPath.find('.', pos + 1))
+            {
+                if(ptrPath[pos - 1] == '\\')
+                {
+                    // Escaped: Replace \. with .
+                    ptrPath.replace(pos - 1, 1, "");
+                }
+                else
+                {
+                    // Replace . with /
+                    ptrPath.replace(pos, 1, "/");
+                }
+            }
```

Comment:
if the string ptrPath starts with a '.', the line `if(ptrPath[pos - 1] == '\')` will access index -1, which is invalid and could lead to undefined behavior.
```suggestion
std::string result;
result.reserve(ptrPath.size());  // To avoid unnecessary relocations
bool prevCharWasSlash = false;

for (char c : ptrPath)
{
    if (c == '.' && !prevCharWasSlash) 
    {
        result += '/';
    }
    else if (c != '\\' || !prevCharWasSlash)
    {
        result += c;
    }
    prevCharWasSlash = (c == '\\');
}
ptrPath = std::move(result) or ptrPath.swap(result)
```

Reply:
Done